File: ruby/calc.rb
# Get the numbers
print "Choose a number: "
num1 = gets.chomp.to_f
print "Choose another number: "
num2 = gets.chomp.to_f

# Get the operation
print "\nOperations:\n1. Addition\n2. Subtraction\n3. Multiplication\n4. Division\n"
print "Choose an operation: "
operation = gets.chomp.to_i

# Parse operation and do the math
if (operation == 1) 
  # addition
  result = num1 + num2
elsif (operation == 2)
  # subtraction
  result = num1 - num2
elsif (operation == 3)
  # multiplication
  result = num1 * num2
elsif (operation == 4)
  # division
  if (num2 == 0)
    puts "You cannot divide by 0!!!"
    exit
  end
  result = num1 / num2
else
  result = "Invalid operation!!!"
end

puts "\nResult: #{result}\n"

File: c/calc.c
#include <stdio.h>

// declare variables
float num1;
float num2;
int operation;
float result;

int main() {
    // define/get the numbers
    printf("Choose a number: ");
    scanf("%f", &num1);
    printf("Choose another number: ");
    scanf("%f", &num2);
    
    // get the operation
    printf("Choose an operation");
    printf("\n 1. Addition \n 2. Subtraction \n 3. Multiplication \n 4. Division \n");
    printf("Choose an operation: ");
    scanf("%d", &operation);

    // parse the operation and actually do math
    if (operation == 1) {
        // addition
        result = num1 + num2;
    } else if (operation == 2) {
        // subtraction
        result = num1 - num2;
    } else if (operation == 3) {
        // multiplication
        result = num1 * num2;
    } else if (operation == 4) {
        // division
        result = num1 / num2;
    } else {
        printf("Invalid operation!!!");
        return 0;
    }
    
    printf("\n");
    printf("Result: %f\n", result);
    
    return 0;
}